File: app/src/main/java/com/example/esl/models/local/entities/Penyewaan.kt
package com.example.esl.models.local.entities

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(tableName = "penyewaan",
    foreignKeys = [
        ForeignKey(
            entity = User::class,
            parentColumns = ["id_users"],
            childColumns = ["id_users"],
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = Property::class,
            parentColumns = ["id_properti"],
            childColumns = ["id_properti"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index(value = ["id_users"]), Index(value = ["id_properti"])]
)
data class Penyewaan(
    @PrimaryKey(autoGenerate = true) val id_penyewaan: Int = 0,
    val id_users: Int,
    val id_properti: Int,
    val tanggalMulai: String,
    val tanggalAkhir: String,
    val tanggalOrder: String,
    val masaSewa: Int,
    val tanggalMulai_Update: String?,
    val tanggalAkhir_Update: String?,
    val status: String, // Enum: Aktif, Selesai, Dibatalkan
    val alasan_batal: String?
)

File: models/penyewaan.js
'use strict';
const {
  Model
} = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class Penyewaan extends Model {
    /**
     * Helper method for defining associations.
     * This method is not a part of Sequelize lifecycle.
     * The `models/index` file will call this method automatically.
     */
    static associate(models) {
      Penyewaan.belongsTo(models.Users, { foreignKey: 'id_users' });
      Penyewaan.belongsTo(models.Properti, { foreignKey: 'id_properti' });
      Penyewaan.hasMany(models.Ulasan, { foreignKey: 'id_penyewaan' });
    }
  }
  Penyewaan.init({
    id_penyewaan: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true,
    },
    id_users: DataTypes.INTEGER,
    id_properti: DataTypes.INTEGER,
    tanggalMulai: DataTypes.DATE,
    tanggalAkhir: DataTypes.DATE,
    tanggalOrder: DataTypes.DATE,
    masaSewa: DataTypes.INTEGER,
    tanggalMulai_Update: DataTypes.DATE,
    tanggalAkhir_Update: DataTypes.DATE,
    status: {
      type: DataTypes.ENUM('Aktif', 'Selesai', 'Dibatalkan'),
      defaultValue:'Aktif'},
    alasan_batal: DataTypes.STRING
  }, {
    sequelize,
    modelName: 'Penyewaan',
  });
  return Penyewaan;
};